legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-dbExtension/legend-engine-xt-relationalStore-snowflake/legend-engine-xt-relationalStore-snowflake-grammar/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/SnowflakeCompilerExtension.java

```diff
@@ -66,6 +69,10 @@ public List<Function2<DatasourceSpecification, CompileContext, Root_meta_pure_al
             if (datasourceSpecification instanceof SnowflakeDatasourceSpecification)
             {
                 SnowflakeDatasourceSpecification snowflakeDatasourceSpecification = (SnowflakeDatasourceSpecification) datasourceSpecification;
+                if (snowflakeDatasourceSpecification.tempTableDb != null ^ snowflakeDatasourceSpecification.tempTableSchema != null)
+                {
+                    throw new EngineException("One of Database name and schema name for temp tables is missing. Please specify both tempTableDb and tempTableSchema", SourceInformation.getUnknownSourceInformation(), EngineErrorType.COMPILATION);
```

Comment:
small change here - should be source info of datasourceSpecification rather than uknown